### ISSUE TITLE
[10.0] FIFO / FEFO 'order by' consumes quants at all locations in parallel due to quants split

### ISSUE THREAD

**Issue by jbaudoux** [odoo/odoo#31186]
FIFO removal strategy is defined as order by 'in_date, id'
FEFO removal strategy is defined as order by 'removal_date, in_date, id'
The "id ASC" must be changed to "id DESC"

Let's assume you have 40 units of product1 in shelf1 and 40 units of product1 in shelf2. They have the same lot and have been received at the same date. So only quant id is different.

Create 2 pickings of 1 unit.

In the current state, the first picking will consume the quant with the lowest id (let's assume on shelf1) and a new quant of 39 units will be created with a new id. The second picking will now consume the quant with the lowest id which is now on shelf2. This means that you always change your source location and never consume entirely one location before consume the next one.
If you have multiple pallets at several locations, you will consume units from all the pallets at the same time instead of emptying one pallet and them go to the next one.

If you change "id asc" by "id desc" then when a quant is split, it will always be that first selected quant that will always be selected.

**Comment by pedrobaeza** [comment]
Version?

**Comment by jbaudoux** [comment]
All versions, including master

**Comment by pedrobaeza** [comment]
@jbaudoux on v11, there's no "quant creation". Only quantities are adjusted, so I don't see the problem for that version. In v10 that can be true (but I haven't analyzed in depth).

**Comment by jbaudoux** [comment]
@pedrobaeza Indeed. I identified that issue in v10. According to the fact that since v11 there is no quant split anymore, the side effect is probably not existing anymore

**Comment by pedrobaeza** [comment]
@sle-odoo can this qualify for a fix only for v10?

**Comment by Yenthe666** [comment]
Dear @jbaudoux,

As Odoo 13 has been released today it means that our support for Odoo version 10 has been stopped.
If you can still reproduce this issue on a newer version (version 11, 12 or 13 at the point of writing) then please do respond back or create a new issue.
Sorry that we didn't get around to resolving your issue.

Regards,
Yenthe
